packages/firestore/src/core/query.ts

```diff
@@ -631,6 +643,24 @@ export class KeyFieldFilter extends FieldFilter {
   }
 }
 
+/** Filter that matches on key fields within an array. */
+export class KeyFieldInFilter extends FieldFilter {
+  constructor(field: FieldPath, op: Operator, value: ArrayValue) {
+    super(field, op, value);
+  }
+
+  matches(doc: Document): boolean {
+    const arrayValue = this.value as ArrayValue;
+    return arrayValue.internalValue.some(refValue => {
+      const comparison = DocumentKey.comparator(
+        doc.key,
+        (refValue as RefValue).key
+      );
+      return this.matchesComparison(comparison);
```

Comment:
I am having trouble seeing how this works...  matchesComparison() is going to use `this.op` to decide what comparison to match for (`<`, `==`, etc.).  But `this.op` is going to be `in` which `matchesComparison()` doesn't support.  So I'm overlooking something or this code isn't being hit by our tests...?

In any case, I _think_ this block could be replaced with something like:
```
return doc.key.isEqual((refValue as RefValue).key);
```

Since all we only need to do an equality check.

Reply:
Thanks for catching another breaking bug. I had validation tests to catch the errors, but not an integration test to check that the end to end query worked. I've added another integration test to check that IN queries actually work with document IDs.